source/ncengine/graphics2/NcGraphicsImpl2.cpp

```diff
@@ -318,24 +359,15 @@ void NcGraphicsImpl2::Run()
 
     auto renderState = m_frontend.BuildRenderState(m_world);
 
-    auto* pRTV = swapChain.GetCurrentBackBufferRTV();
-    auto* pDSV = swapChain.GetDepthBufferDSV();
-    context.SetRenderTargets(1, &pRTV, pDSV, Diligent::RESOURCE_STATE_TRANSITION_MODE::RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
-
-    constexpr auto ClearColor = Vector4{0.050f, 0.050f, 0.050f, 1.0f};
-    context.ClearRenderTarget(pRTV, &ClearColor.x, Diligent::RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
-    context.ClearDepthStencil(pDSV, Diligent::CLEAR_DEPTH_FLAG, 1.f, 0, Diligent::RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
-
-    m_postProcessPassBackend.Update(context, renderState.postProcessState);
+    m_passBackend.Update(renderState.postProcessState);
     m_shaderBindings.Update(context, device, renderState);
     m_shaderBindings.GetPerFrameSignature().Commit(context);
+    m_shaderBindings.GetPerPassSignature().Commit(context);
```

Comment:
Just doing a quick pass now, but if its a 'per pass' resource, wouldn't we want to commit later?

Reply:
No - commit is only needed once per frame and additionally once whenever the underlying resource (buffer, texture) on the GPU that the SRB is pointing to changes. We don't currently change the pointers to underlying resources outside of this initial set here.

The PerPass distinction is still required because we need to call `TransitionShaderResources` at SRB scope for each pass (it transitions the render targets from `RenderTargetView` to `ShaderResourceView`